control-plane/api/mesh/v2beta1/api_gateway_types.go

```diff
@@ -43,13 +42,13 @@ type APIGateway struct {
 }
 
 type APIGatewayStatus struct {
-	Status
-	Addresses []GatewayAddress
-	Listeners []ListenerStatus
+	Status    `json:"status,omitempty"`
+	Addresses []GatewayAddress `json:"addresses"`
+	Listeners []ListenerStatus `json:"listeners"`
 }
 
 type ListenerStatus struct {
-	Status
+	Status         `json:"status,omitempty"`
```

Comment:
@sarahalsmiller I updated these with json tags, not sure if the tags are correct but I couldn't run `make ctrl-manifests` without them